### ISSUE TITLE
Jest Testing: Error [TypeError]: ModelStore is not a function [BUG]

### ISSUE THREAD

**Issue by mrp67** [dynamoose/dynamoose#1144]
### Summary:
While running Jest tests that initialize a model with a large schema, I am encountering the following error:

Error [TypeError]: ModelStore is not a function

This error occurs in normal and runInBands modes. I am unable to reproduce it in a sample project/snippet.

I found the source of the issue and have a fix outlined below.

### Current output and behavior (including stack trace):

Error [TypeError]: ModelStore is not a function
    at new Model (/code/node_modules/dynamoose/lib/Model/index.ts:398:3)

### Expected output and behavior:

The expected behavior is that this error would not occur. It seems to be caused by a require statement in the Model constructor source code.
https://github.com/dynamoose/dynamoose/blob/aa22c7e1ee7484b58fdd1624ff2b28a5b5567ae7/lib/Model/index.ts#L397-L398

My assumption is that in certain scenarios, the call to ModelStore is executed before the require statement returns a value.

If this require statement is moved to the top of the file with the other require statements, the issue resolves itself and ModelStore is able to be called as expected.

### Environment:

Operating System: macOS (My code is running in an alpine linux v3.13.2 docker image)
Operating System Version: 10.15.7
Node.js version (`node -v`): 14.15.4
NPM version: (`npm -v`): 6.14.11
Dynamoose version: 2.7.1 (this issue existed in prior versions too)

### Other:
- [x] I have read through the Dynamoose documentation before posting this issue
- [x] I have searched through the GitHub issues (including closed issues) and pull requests to ensure this issue has not already been raised before
- [x] I have searched the internet and Stack Overflow to ensure this issue hasn't been raised or answered before
- [x] I have tested the code provided and am confident it doesn't work as intended
- [x] I have filled out all fields above
- [x] I am running the latest version of Dynamoose

**Comment by fishcharlie** [comment]
@mrp67 Ok this is a complicated thing. Lots of thoughts here.

First a few questions.

1. Is this reproducible 100% of the time?
2. What version of Dynamoose did this start occurring in?
3. Does this only occur with Jest? Could Jest be modifying `require` somehow to break this? If so, it's likely an issue with Jest.

---

1. The fact that you are unable to reproduce it in an example project is concerning. Really makes me wonder if there is truly something wrong with Dynamoose here or if it's an issue on your side.
2. `the call to ModelStore is executed before the require statement returns a value`. I can see why you think that, **but** it goes against everything that `require` statements are based on. `require` statements are synchronous, that is just one of the foundational rules about them. If they weren't, there would be a callback or promise returned to determine when it's loaded.
3. `If this require statement is moved to the top of the file with the other require statements`. I think originally we had a dependency cycle here, which was causing problems. Since then there is a chance changes have been made to eliminate that.

---

Not really sure what to say here. If you want to submit a PR and it passes all the tests I'd be happy to review it. But other than that, without a reproducible example or more information, it's **really** hard to say what next steps here. I **really** avoid submitting PRs or making changes myself unless I can reproduce the bug myself. So if you would like to, feel free.

I think ideally we'd get to the deeper problem here before moving forward. If that require statement is truly async, we really need to figure out why.

**Comment by mrp67** [comment]
@fishcharlie 
This started occurring at least since 2.7.0

I think I've been able to narrow it down a bit further since opening this issue. It appears to be an issue within my work environment. Some async code was added to the tests which was causing Jest to freak out. I'm not sure why it was targeting the dynamoose model, specifically the require of ModelStore. 

I did clone dynamoose and move the require statement. All tests passed. 
Since the problem appears to not be with dynamoose itself, I'll close this issue and continue to keep an eye on things.
